### ISSUE TITLE
Questionable Eq and Ord instances for CReal

### ISSUE THREAD

**Issue by expipiplus1** [jwiegley/numbers#17]
I think functions to compare equality should be exposed (with a digits-to-compare parameter), but exposing CReal as an instance of Eq and Ord is a little fishy, as this can break a whole bunch of the informal laws surrounding Haskell's Eq and Ord classes.

Although other types such as Float and Double break the laws of Eq, I think it would be appropriate to hold "exact" real arithmetic to a higher standard. 

```
Data.Number.CReal> 2e-42 == (1e-42 :: CReal)
True
```

```
 Data.Number.CReal> 2e-42 > (1e-42 :: CReal)
False
```

**Comment by cartazio** [comment]
yeah ... Eq and Ord for CReal style numbers needs to have an "epsilon" or "digit" argument

**Comment by expipiplus1** [comment]
@cartazio you might be interested in my 'exact-real' package!
